### PR TITLE
Classify transition lists by first attempting to read them as proteomics transition lists

### PR DESCRIPTION
**Pull request by henrytsanford**
Previously we would search for headers particular to small molecule transition lists in order to determine if a transition list was peptide or small molecule. With this change, we try to create a peptide row reader using the first 100 lines of the pasted text. In order to create a peptide row reader, we must find the matching amino acid sequence and precursor m/z columns, among other things. If we can successfully create a peptide row reader, we assume it is a peptide transition list. 

If we are unable to create a peptide row reader, it falls back on our existing logic and looks at the headers. 

This also fixes a small redundancy where the same transition list would be classified as peptide or small molecule twice in the course of one import.

### CODE REVIEW

File: pwiz_tools/Skyline/Model/Import.cs [hunk 1]
```diff
@@ -373,29 +373,35 @@ private void InitFormat(IList<string> inputLines)
         {
             if (FormatProvider == null)
             {
-                char sep;
-                IFormatProvider provider;
-                Type[] columnTypes;
-                string inputLine = 0 < inputLines.Count ? inputLines[0] : string.Empty;
-                if (!MassListImporter.IsColumnar(inputLine, out provider, out sep, out columnTypes))
+                if (!TryInitFormat(inputLines, out var provider, out var sep))
```
Comment (bspratt) [inline pwiz_tools/Skyline/Model/Import.cs:376]: Could probably use some comments through here.

File: pwiz_tools/Skyline/Model/Import.cs [hunk 2]
```diff
@@ -412,19 +418,31 @@ public class MassListImporter
 
         private int _linesSeen;
 
-        public MassListImporter(SrmDocument document, MassListInputs inputs)
+        // This constructor is only suitable for investigating the peptide-vs-small molecule nature of inputs
+        // CONSIDER(henryS) Arguably that code should be split out into its own class
```
Comment (bspratt) [inline pwiz_tools/Skyline/Model/Import.cs:422]: Brendan, is it worth the churn?

File: pwiz_tools/Skyline/Model/Import.cs [hunk 3]
```diff
@@ -412,19 +418,31 @@ public class MassListImporter
...
 
         public SrmDocument Document { get; private set; }
         public MassListRowReader RowReader { get; private set; }
-        public SrmSettings Settings { get { return Document.Settings; } }
+        public SrmSettings Settings { get; private set; }
         public MassListInputs Inputs { get; private set; }
         public IFormatProvider FormatProvider { get { return Inputs.FormatProvider; } }
         public char Separator { get { return Inputs.Separator; } }
-        public bool IsSmallMoleculeInput { get; private set; }
+        public SrmDocument.DOCUMENT_TYPE InputType { get; private set; }
```
Comment (bspratt) [inline pwiz_tools/Skyline/Model/Import.cs:445]: comment this, something like // Peptide vs Smallmol vs None (meaning Unknown)

File: pwiz_tools/Skyline/Model/SrmDocument.cs [hunk 4]
```diff
@@ -1482,11 +1482,13 @@ private SrmDocument MergeMatchingPeptidesUserInfo(IList<PeptideGroupDocNode> pep
             return docNew;
         }
 
-        public MassListImporter PreImportMassList(MassListInputs inputs, IProgressMonitor progressMonitor, bool tolerateErrors)
+        public MassListImporter PreImportMassList(MassListInputs inputs, IProgressMonitor progressMonitor, bool tolerateErrors, DOCUMENT_TYPE inputType = DOCUMENT_TYPE.none)
```
Comment (bspratt) [inline pwiz_tools/Skyline/Model/SrmDocument.cs:1485]: a little explanatory comment header here would be nice

File: pwiz_tools/Skyline/SkylineFiles.cs [hunk 5]
```diff
@@ -1871,7 +1871,7 @@ public void ImportMassList(string fileName)
             }
         }
 
-        public void ImportMassList(MassListInputs inputs, string description, bool assayLibrary)
+        public void ImportMassList(MassListInputs inputs, string description, bool assayLibrary, SrmDocument.DOCUMENT_TYPE inputType = SrmDocument.DOCUMENT_TYPE.none)
```
Comment (bspratt) [inline pwiz_tools/Skyline/SkylineFiles.cs:1874]: General need for comment headers throughout, for the next coder that comes along

**Review by bspratt — CHANGES_REQUESTED** [review]
It strikes me (and this was true before your changes) that there's an awful lot of logic in EditMenu.cs that would be useful to the commandline version of Skyline. We should probably investigate that and unify the code. That is, move that logic to Import.cs, probably.

Otherwise see individual review comments, which are mostly about wanting more comments.

File: pwiz_tools/Skyline/Model/SrmDocument.cs (hunk 4 above)
Comment (henrytsanford) [reply pwiz_tools/Skyline/Model/SrmDocument.cs:1493]: I added a comment describing the method and the parameter I added. Is this the comment format you are looking for? I am not sure what the best comment style is when I only want to comment on one of the parameters.

Comment (bspratt) [reply pwiz_tools/Skyline/Model/SrmDocument.cs:1493]: If you don't want to go the full Monty on the comment, then just put it on its own line and comment that:
       /// <summary>
        /// Return a mass list import if the progress monitor is not cancelled and we are able to read the document
        /// </summary>
       public MassListImporter PreImportMassList(MassListInputs inputs, IProgressMonitor progressMonitor, bool tolerateErrors,
            DOCUMENT_TYPE inputType = DOCUMENT_TYPE.none) // "None" means "don't know if it's peptides or small molecules, go figure it out".

**Review by bspratt — APPROVED** [review]
OK, you've verified that there's not an issue with model and UI code intermingling. Sounds like Brendan is OK with getting this into master.

**Pull request merged**